### PR TITLE
Preprocess doc body & regions in generateISD

### PR DESCRIPTION
**Pull request by jlks**
This is a performance enhancement for the `generateISD` function. 

The content of the body is processed once, creating a copy containing only the content with timings to fit the offset provided. A set of regions used by this content is also created and then used when actually creating the ISD instead of processing every region defined.

Previously this function was processing the entire body per region.

### CODE REVIEW

**Comment by nigelmegitt** [comment]
Note for whoever merges this: please Squash.

**Review by nigelmegitt — APPROVED** [review]
Enormous speed-up! Quick test of the output reveals no any ill-effects that I could see.

I think the library has a bug in the handling of default regions: it does not seem to pass https://github.com/w3c/ttml2-tests/blob/master/presentation/valid/ttml2-prstn-region-no-default.xml but this was the same before and after this change. I also constructed a local test to check that the default region is used when no `region` element is specified, and this passes both before and after this change.

**Comment by nigelmegitt** [comment]
Just to add, the default region behaviour is not of any concern to us, whether it passes or fails the test I linked to above, since we don't have any content that would be affected.

File: BBC-CHANGELOG.md [hunk 1]
```diff
@@ -7,4 +7,4 @@ For more detailed changes, please use the commit history.
 ## Changes
 
 - Handle TTML1 1st Edition CR ttaf namespace declarations
-
+- Performance improvements for `generateISD`. [#6](https://github.com/bbc/imscJS/pull/6)
```
Comment (domjacks) [inline BBC-CHANGELOG.md]: 👍

**Review by domjacks — APPROVED** [review]

**Review by robertbryer — APPROVED** [review]

**Pull request merged**